test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1

```diff
@@ -64,3 +64,73 @@ Describe "Native Command Arguments" -tags "CI" {
         }
     }
 }
+
+Describe 'PSPath to native commands' {
+    BeforeAll {
+        $featureEnabled = $EnabledExperimentalFeatures.Contains('PSNativePSPathResolution')
+
+        if ($IsWindows) {
+            $cmd = "cmd"
+            $cmdArg1 = "/c"
+            $cmdArg2 = "type"
+            $dir = "cmd"
+            $dirArg1 = "/c"
+            $dirArg2 = "dir"
+        }
+        else {
+            $cmd = "cat"
+            $dir = "ls"
+        }
+
+        Set-Content -Path testdrive:/test.txt -Value 'Hello'
+        Set-Content -Path "testdrive:/test file.txt" -Value 'Hello'
+        Set-Content -Path "env:/test var" -Value 'Hello'
+        $filePath = Join-Path -Path ~ -ChildPath (New-Guid)
+        Set-Content -Path $filePath -Value 'Home'
+    }
+
+    AfterAll {
+        Remove-Item -Path "env:/test var"
+        Remove-Item -Path $filePath
+    }
+
+    It 'PSPath with ~/path works' -Skip:(-not $featureEnabled) {
+        $out = & $cmd $cmdArg1 $cmdArg2 $filePath
+        $LASTEXITCODE | Should -Be 0
+        $out | Should -BeExactly 'Home'
+    }
+
+    It 'PSPath with ~ works' -Skip:(-not $featureEnabled) {
+        $out = & $dir $dirArg1 $dirArg2 ~
+        $LASTEXITCODE | Should -Be 0
+        $out | Should -Not -BeNullOrEmpty
+    }
+
+    It 'PSPath that is file system path works with native commands: <path>' -Skip:(-not $featureEnabled) -TestCases @(
```

Comment:
It would be good to add tests for drive names and paths with uncommon (but allowed) characters.
E.g. `New-PSDrive -Name "My test! ;+drive" -Root C:\1 -PSProvider FileSystem`

Reply:
Will add.  This works, but found an issue where `temp:test.txt` doesn't work as expected (vs `temp:/test.txt`) will make a fix.